airflow/providers/databricks/operators/databricks.py

```diff
@@ -285,12 +285,12 @@ def __init__(
         *,
         json: Any | None = None,
         tasks: list[object] | None = None,
-        spark_jar_task: dict[str, str] | None = None,
-        notebook_task: dict[str, str] | None = None,
-        spark_python_task: dict[str, str | list[str]] | None = None,
-        spark_submit_task: dict[str, list[str]] | None = None,
-        pipeline_task: dict[str, str] | None = None,
-        dbt_task: dict[str, str | list[str]] | None = None,
```

Comment:
The current typing is sufficient IMO. These args would eventually be of these types either from XComs, templates, etc.;`dict[str, object]` is nebulous.

Reply:
Agreed, please revert these lines.

Reply:
Ok, this might be unrelated to the specific issue we are mentioning here but passing a `dict[str, dict[str, str]]` is also accepted and produces expected outcome under `notebook_task` parameter, for example.

However, when running mypy checks, it fails. I agree that maybe passing as `object` might be vague and some more thought will be needed but, at the very least, the key for `notebook_task` param (and probably for the rest too) should accept `dict`, `str`, `int` to go in line with what's there so far (i.e., without this PR) and, eventually `XComArg` and `PlainXComArg` to support the changes in this PR

Reply:
I may also be a good idea to split the changes into two PRs, one does not need to block the other.

Reply:
Awesome! Let's do that then. I am gonna revert those changes and then submit a new PR for the other one once this is merged.